Added the possibility to not generate iat

nacron opened the pull request:
Hello,
I had the problem that the iat was causing a problem with clock skew and jwt checks.
I added the possibility to not create the iat. 
It is basically passing the option noTimestamp (true / false) to jwt.sign().

Cheers
Benedikt

nacron committed:
Added the possibility to not generate iat

mjamado commented:
Hi, @nacron, thank you for your PR.

However, I can't accept it in its current state. Changing the order of the parameters, like you did by inserting a new one on the middle, will break current installs for everyone who upgrades.

New parameters must be added at the end. If that's acceptable to you, please update your PR.

As far as removing the IAT - it's optional, but it's inclusion is a security feature. If one or more of your servers are skewing, either don't check the IAT on arrival or, which I heartly recommend, fix the skew at the OS level using NTP or something.

nacron commented:
Hey @mjamado,

Understood the ordering of parameter. Was wondering why it wasn't working with my old plugin. I am getting right on it.
About the clock skew, my problem is not the server but my client and the shitty ntp server in my company network. I have the feeling that could happen to some other people as well...

nacron committed:
Changes from Pull Request

nacron commented:
I changed the order now. I also updated the tests.
I also changed the default so IAT is created.
Tested it with my old configurations and it worked.
